libraries/chain/controller.cpp

```diff
@@ -1318,6 +1366,14 @@ struct controller_impl {
 
                db.commit( (*bitr)->block_num() );
                root_id = (*bitr)->id();
+
+               if (should_transition_to_savanna(*bitr)) {
+                  savanna_transistion_required = true;
+                  // Do not advance irreversible past IF Genesis Block
+                  break;
+               } else if ((*bitr)->block->is_proper_svnn_block()) {
+                  fork_db.switch_to_savanna();
```

Comment:
Why do we have two similar functions `fork_db.switch_to_savanna()` and `fork_db.switch_from_legacy()`?

Reply:
They are used at the different stages of transition.